### PR TITLE
Scala/Java Predict API fix #14756

### LINKED ISSUE #14756: Mismatch between shape Java API
Hi, I’ve finished training the yolo3_darknet53 params and wanted to test it with Java API, however am having this error which I have no idea where it coming from:

     Exception in thread “main” java.lang.IllegalArgumentException: requirement failed:
     Mismatch between shape (100,1) and (1,100,4)
        	at scala.Predef$.require(Predef.scala:224)
        	at org.apache.mxnet.NDArray$$anonfun$9.apply(NDArray.scala:448)
        	at org.apache.mxnet.NDArray$$anonfun$9.apply(NDArray.scala:445)
        	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
        	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
        	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
        	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
        	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
        	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
        	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
        	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
        	at org.apache.mxnet.NDArray$.concatenate(NDArray.scala:445)
        	at org.apache.mxnet.module.BaseModule$$anonfun$2.apply(BaseModule.scala:267)
        	at org.apache.mxnet.module.BaseModule$$anonfun$2.apply(BaseModule.scala:267)
        	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
        	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
        	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
        	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
        	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
        	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
        	at org.apache.mxnet.module.BaseModule.predict(BaseModule.scala:267)
        	at org.apache.mxnet.infer.Predictor$$anonfun$11.apply(Predictor.scala:210)
        	at org.apache.mxnet.infer.Predictor$$anonfun$11.apply(Predictor.scala:210)
        	at org.apache.mxnet.infer.MXNetThreadPoolHandler$$anon$4.call(MXNetHandler.scala:73)
        	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        	at java.lang.Thread.run(Thread.java:748)
    FAILURE: Build failed with an exception.

    * What went wrong:
    Execution failed for task ':ObjectDetectionTutorial.main()'.
    > Process 'command '/usr/lib/jvm/java-8-openjdk-amd64/bin/java'' finished with non-zero exit value 1

    * Try:
    Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

    * Get more help at https://help.gradle.org

    BUILD FAILED in 1s
    2 actionable tasks: 2 executed
    Process 'command '/usr/lib/jvm/java-8-openjdk-amd64/bin/java'' finished with non-zero exit value 1
    10:16:24: Task execution finished 'ObjectDetectionTutorial.main()'.

I only have 1 class and my input image is 512 and pretrained_base=false. Any help would be much appreciated. Thank you in advance.

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##
A bug are being identifed that the concatenation process applied on the predict is incorrect. This PR is a fix on that #14756.
@zachgk @gigasquid @yzhliu @androuino

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: scala-package/core/src/main/scala/org/apache/mxnet/module/BaseModule.scala [hunk 1]
```diff
@@ -264,7 +276,8 @@ abstract class BaseModule {
           s"in mini-batches (${out.size})." +
       "Maybe bucketing is used?")
     )
-    val concatenatedOutput = outputBatches.map(out => NDArray.concatenate(out))
+    val oBT = outputBatches.transpose
+    val concatenatedOutput = oBT.map(out => NDArray.concatenate(out))
```
Comment (piyushghai) [inline scala-package/core/src/main/scala/org/apache/mxnet/module/BaseModule.scala]: Do you also need to dispose ```oBT``` here ?

File: scala-package/core/src/main/scala/org/apache/mxnet/module/BaseModule.scala [hunk 2]
```diff
@@ -247,6 +247,18 @@ abstract class BaseModule {
 
   /**
    * Run prediction and collect the outputs.
+   * The concatenation process will be like
+   * {{{
+    *   outputBatches = [
+    *      [a1, a2, a3], // batch a
+    *      [b1, b2, b3]  // batch b
+    *   ]
+    *   result = [
```
Comment (piyushghai) [inline scala-package/core/src/main/scala/org/apache/mxnet/module/BaseModule.scala]: Can you add this to the ```@return``` docs for this method ?

File: scala-package/core/src/main/scala/org/apache/mxnet/module/BaseModule.scala (hunk 1 above)
Comment (lanking520) [reply scala-package/core/src/main/scala/org/apache/mxnet/module/BaseModule.scala]: This create a soft copy, the content inside contains same reference
